internal/cli/clients/client.go

```diff
@@ -1725,6 +1700,10 @@ func uniqueStrings(stringSlice []string) []string {
 }
 
 func getAPI(ctx context.Context, log logr.Logger) (string, string, error) {
+	// This is called only by the admin command `config update`
+	// which has to talk to the cluster to retrieve the
+	// information. This is allowed.
```

Comment:
maybe it would be safer to have a different "admin client" class that we can use when kube access might be needed. This way it will be harder to accidentally introduce commands that access the kube api when they shouldn't. I think we can rely on code reviews to avoid that for now but if we fail we should think about extracting the functionality to another client.

Reply:
Agreed, while also out of scope for now. Will make a ticket for this tech debt.

Reply:
See #719